### DISCUSSION TITLE
Long-distance WAN Latency Performance Problem

### DISCUSSION THREAD

**Discussion by Akyer3** [gpiozero/gpiozero#1050] [Q&A] [answered]
Hi. I am new when it comes to... this whole business in general actually. One day I wanted to collaborate with my friend who lives halfway across the world (I live in Korea, and my friend lives in Georgia, US). Please forgive me if I made a mistake, or fail to understand what is going on. You could assume that I know only the baby-basics.

My friend and I have constructed a wheeled robot with Raspberry Pi Zero W. We both had a solid internet connection. We thought it would be fun to control each other robots and Raspberry Pi during video calls. Our latency when making a Discord video call is about 40-50 ms. This low latency prompts me to believe that simple commands sent with the gpiozero library could easily achieve rather low latency.

It is to my surprise that the latency from traveling from my friend (who is controlling the robot, "Master") to my own robot ("Slave") is no less than 3 full seconds. A surprise considering that my friend who ran an internet speed test online connecting to a server in Seoul, SK has gotten a latency of about 200 ms.

I have performed my own personal test at a much much closer distance. I have created (or at least I believe have) two separate LANs at home where the control PC has to connect to my Raspberry Pi robot using a public IP address. The latency between my home PC and the Raspberry Pi is not noticeable here.

Is there a way to decrease this long-distance latency? Why do we have a full 3+ second delay, when using Discord or using the online latency test achieves a much faster result? I would like to achieve a latency of less than a full second (1000ms), ideally >200ms if possible.
Thank you for reading. This library has been really fun to use.

**Comment by lurch** [comment, marked as answer]
I guess it depends exactly how you've set up your routing / firewall-rules / etc. to allow you to access devices on each other's networks? Sounds like a general networking problem rather than anything specific to GPIO Zero? (so perhaps better asked on https://forums.raspberrypi.com/ ?)
What response times do you get when you try doing a command-line ping of the IP address of each other's robots?  Also worth bearing in mind that some routers probably use QoS to give higher-priority to video-call traffic than other types of network traffic.

**Reply by Akyer3** [reply]
Sorry for replying late
You were right. The problem wasn't the GPIO Zero library, but with the networking.

After organizing a meetup with my buddy, we tried what you recommend and used a tracert command to test out the latency response time between him and I. Pinging from Georgia, USA to Seoul, South Korea took more than a combined 1500ms+ response time with one server in particular taking up a huge 644ms. This server was sending the ping from Chicago to Seoul.

This high delay confirmed what you said; some traffic were given a higher priority to reach to it's destination faster. My guess is that maintaining such high speed cost a lot of money.

I have no idea how to make it so that I could use this high speed priority network traffic. But this question is outside the bound of this library.
Thank you so much for replying and helping me out.
